sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -144,7 +144,7 @@ class Analyzer(
 
   lazy val batches: Seq[Batch] = Seq(
     Batch("Hints", fixedPoint,
-      new ResolveHints.ResolveBroadcastHints(conf),
+      new ResolveHints.ResolveBroadcastHints(conf, catalog),
```

Comment:
`catalog` is too big for this since we only use the current database name.
```scala
-      new ResolveHints.ResolveBroadcastHints(conf, catalog),
+      new ResolveHints.ResolveBroadcastHints(conf, catalog.getCurrentDatabase),
```

Reply:
aha, ok.